src/node/structural_hash.cc

```diff
@@ -360,6 +361,39 @@ struct ADTObjTrait {
 
 TVM_REGISTER_REFLECTION_VTABLE(runtime::ADTObj, ADTObjTrait);
 
+struct ModuleNodeTrait {
+  static constexpr const std::nullptr_t VisitAttrs = nullptr;
+  static void SHashReduce(const runtime::ModuleNode* key, SHashReducer hash_reduce) {
```

Comment:
avoid structural equality check of runtime module as it is not guaranteed to be valid for all. In this case let us avoid use structural equality for check and instead do some special handling

Reply:
> avoid structural equality check of runtime module as it is not guaranteed to be valid for all.

Is it because runtime module is not always binary serializable? If so, current design would raise an error within the structural equality. I assumed the serialization is always deterministic.

Reply:
Based on offline discussion and https://github.com/apache/tvm/pull/15244#discussion_r1294883567, we agreed to disable it for now and revisit when needed in the future.